src/theory/quantifiers/index_trie.h

```diff
@@ -0,0 +1,76 @@
+/*********************                                                        */
+/*! \file index_trie.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Mikolas Janota
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Implementation of a trie that store subsets of tuples of term indices
+ ** that are not yielding  useful instantiations. of quantifier instantiation.
+ ** This is used in the term_tuple_enumerator.
+ **/
+#ifndef INDEX_TRIE_H_45
+#define INDEX_TRIE_H_45
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+#include "base/check.h"
+
+namespace CVC4 {
```

Comment:
Our namespaces (mostly) follow directory structure, so this should use 3 nested namespace, e.g. https://github.com/ajreynol/CVC4/blob/master/src/theory/quantifiers/alpha_equivalence.h#L24